components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1438,60 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, Unit<Length> unit) {
+    if (value == null || value.isInfinite()) {
+      LOGGER.debug("Expected float value for position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) unit = UNITS.REFERENCEFRAME;
```

Comment:
Why the hardcoding of reference frame as a fallback?  This is a situation where it would be appropriate to throw an exception.  Note that for every model property, we *know* the default unit type and it's reasonable to require a valid unit here.

As an example, this is the C++ `update()` logic for `Pixels/PhysicalSizeY`

```
        if (element.hasAttribute("PhysicalSizeY"))
          {
            // Attribute property PhysicalSizeY with unit companion PhysicalSizeYUnit
            std::string unitSymbol("µm");
            if (element.hasAttribute("PhysicalSizeYUnit"))
              {
                unitSymbol = element.getAttribute("PhysicalSizeYUnit");
                if(unitSymbol.empty())
                  unitSymbol = "µm";
              }

            set_quantity(element.getAttribute("PhysicalSizeY"), unitSymbol,
                         *this, &Pixels::setPhysicalSizeY,
                         "Pixels", "PhysicalSizeY");
          }
```

Here we default to µm if unspecified in the input.  But for any *API* called by a *reader*, the reader itself must already know the unit system it's dealing with.  If it doesn't this it must itself pass in reference frame as the unit type.  I don't think this type of hardcoded fallback is appropriate.  This is the reader's responsibility.